src/components/item/sharing/ItemSharingTab.tsx

```diff
@@ -33,6 +34,28 @@ type Props = {
   memberships?: ItemMembership[];
 };
 
+interface PermissionMap {
+  [key: string]: ItemMembership;
+}
+const reduceMembershipsForSameMember = (
+  memberships: ItemMembership[],
+): ItemMembership[] => {
+  const permissionMap: PermissionMap = {};
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const membership of memberships) {
+    const { member, permission } = membership;
+
+    if (
+      !permissionMap[member.id] ||
+      PermissionLevelCompare.gt(permission, permissionMap[member.id].permission)
+    ) {
+      permissionMap[member.id] = membership;
+    }
+  }
```

Comment:
You can optimize with a `.reduce` function. I guess the accumulator could look like `{...acc, [member.id]: membership }`